File: branches/0791/PokemonData/Custom/UpdatePokemonData.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PokemonBattle.PokemonData.Custom
{
    [Serializable()]
    public class UpdatePokemonData
    {
        public int Identity
        { get; set; }

        public string NameBase
        { get; set; }

        public double Weight
        { get; set; }

        public int Number
        { get; set; }

        public byte HPBase
        { get; set; }

        public byte AttackBase
        { get; set; }

        public byte DefenceBase
        { get; set; }

        public byte SpeedBase
        { get; set; }

        public byte SpAttackBase
        { get; set; }

        public byte SpDefenceBase
        { get; set; }

        public int Type1
        { get; set; }

        public int Type2
        { get; set; }

        public Trait Trait1
        { get; set; }

        public Trait Trait2
        { get; set; }

        public List<int> AddMoves
        { get; private set; }

        public List<int> RemoveMoves
        { get; private set; }

        public UpdatePokemonData()
        {
            AddMoves = new List<int>();
            RemoveMoves = new List<int>();
            Type1 = PokemonType.InvalidId;
            Type2 = PokemonType.InvalidId;
        }

        public UpdatePokemonData Clone()
        {
            UpdatePokemonData data = MemberwiseClone() as UpdatePokemonData;
            data.AddMoves = new List<int>(AddMoves);
            data.RemoveMoves = new List<int>(RemoveMoves);
            return data;
        }

        public bool Equals(UpdatePokemonData data)
        {
            if (data == null) return false;
            if (data.Identity != Identity) return false;
            if (data.NameBase != NameBase) return false;
            if (data.Weight != Weight) return false;
            if (data.Number != Number) return false;

            if (data.HPBase != HPBase) return false;
            if (data.AttackBase != AttackBase) return false;
            if (data.DefenceBase != DefenceBase) return false;
            if (data.SpeedBase != SpeedBase) return false;
            if (data.SpAttackBase != SpAttackBase) return false;
            if (data.SpDefenceBase != SpDefenceBase) return false;

            if (data.Type1 != Type1) return false;
            if (data.Type2 != Type2) return false;
            if (data.Trait1 != Trait1) return false;
            if (data.Trait2 != Trait2) return false;

            if (data.AddMoves.Count != AddMoves.Count) return false;
            for (int i = 0; i < data.AddMoves.Count; i++)
            {
                if (data.AddMoves[i] != AddMoves[i]) return false;
            }
            if (data.RemoveMoves.Count != RemoveMoves.Count) return false;
            for (int i = 0; i < data.RemoveMoves.Count; i++)
            {
                if (data.RemoveMoves[i] != RemoveMoves[i]) return false;
            }
            return true;
        }
    }
}

File: branches/0791/Pokemon/DataClass/CustomData/UpdatePokemonData.vb
<Serializable()> Public Class UpdatePokemonData

    Private _nameBase As String

    Private HPBase, AttackBase, DefenceBase, SpeedBase, SpAttackBase, SpDefenceBase As Byte
    Private weight As Double
    Private type1, type2 As String

    Private trait1, trait2 As Integer

    Private addMoveList As List(Of String)
    Private removeMoveList As List(Of String)
    Public Sub New(ByVal identity As Integer, ByVal pokemonName As String)
        _nameBase = pokemonName
        _identity = identity
        type1 = ""
        type2 = ""
        addMoveList = New List(Of String)
        removeMoveList = New List(Of String)
    End Sub

    Public ReadOnly Property Name() As String
        Get
            Return _nameBase
        End Get
    End Property
    Public Property NewHPBase() As Byte
        Get
            Return HPBase
        End Get
        Set(ByVal value As Byte)
            HPBase = value
        End Set
    End Property
    Public Property NewAttackBase() As Byte
        Get
            Return AttackBase
        End Get
        Set(ByVal value As Byte)
            AttackBase = value
        End Set
    End Property
    Public Property NewDefenceBase() As Byte
        Get
            Return DefenceBase
        End Get
        Set(ByVal value As Byte)
            DefenceBase = value
        End Set
    End Property
    Public Property NewSpeedBase() As Byte
        Get
            Return SpeedBase
        End Get
        Set(ByVal value As Byte)
            SpeedBase = value
        End Set
    End Property
    Public Property NewSpAttackBase() As Byte
        Get
            Return SpAttackBase
        End Get
        Set(ByVal value As Byte)
            SpAttackBase = value
        End Set
    End Property
    Public Property NewSpDefenceBase() As Byte
        Get
            Return SpDefenceBase
        End Get
        Set(ByVal value As Byte)
            SpDefenceBase = value
        End Set
    End Property
    Public Property NewWeight() As Double
        Get
            Return weight
        End Get
        Set(ByVal value As Double)
            weight = value
        End Set
    End Property

    Public Property NewType1() As String
        Get
            Return type1
        End Get
        Set(ByVal value As String)
            type1 = value
        End Set
    End Property
    Public Property NewType2() As String
        Get
            Return type2
        End Get
        Set(ByVal value As String)
            type2 = value
        End Set
    End Property
    Public Property NewTrait1() As Integer
        Get
            Return trait1
        End Get
        Set(ByVal value As Integer)
            trait1 = value
        End Set
    End Property
    Public Property NewTrait2() As Integer
        Get
            Return trait2
        End Get
        Set(ByVal value As Integer)
            trait2 = value
        End Set
    End Property

    Private _identity As Integer
    Public ReadOnly Property Identity() As Integer
        Get
            Return _identity
        End Get
    End Property

    Public ReadOnly Property AddMove() As List(Of String)
        Get
            Return addMoveList
        End Get
    End Property
    Public ReadOnly Property RemoveMove() As List(Of String)
        Get
            Return removeMoveList
        End Get
    End Property

    Public Function Clone() As UpdatePokemonData
        Dim obj As UpdatePokemonData = CType(Me.MemberwiseClone, UpdatePokemonData)
        obj.addMoveList = New List(Of String)
        obj.addMoveList.AddRange(addMoveList)
        obj.removeMoveList.AddRange(removeMoveList)
        Return obj
    End Function
End Class
